test/lit/nominal-no-gc.wast

```diff
@@ -0,0 +1,15 @@
+;; Write the module with --nominal but without GC
+;; RUN: wasm-opt %s --nominal --disable-gc -g -o %t.wasm
+
+;; We should not get any recursion groups even though we used --nominal
+;; RUN: wasm-opt %t.wasm --hybrid -all -S -o - | filecheck %s
```

Comment:
The `--hybrid` is load-bearing here. Without it, we would just ignore any rec groups in the binary and the test would pass even with the bug present. `-all` similarly might be necessary to allow any rec groups in the binary to reach the output so that the test fails correctly. I'll mention this in the comment.

Reply:
Hmm, but it errors if rec groups are present but gc is disabled - that's the original error that led to this PR? I just want to make sure we have testing for that specifically. How about testing both paths?

Reply:
IIUC, It's wasm2c that errors in the presence of recursion groups, not binaryen.

Reply:
Binaryen errors too:
```wat
(module
  (func $foo)
)
```
```
$ wasm-opt -all --nominal a.wat -o a.wasm
warning: no passes specified, not doing any work
$ bin/wasm-opt a.wasm
[parse exception: Recursion groups not allowed with equirecursive typing (at 0:13)]
Fatal: error parsing wasm
```

Reply:
Ok, I added another run line that tests that we don't get a failure loading the module with the default flags.